core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -4595,6 +4595,184 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
     }
   }
 
+  test("SPARK-40082: recomputation of shuffle map stage with no pending partitions" +
+    "should not hang") {
```

Comment:
nit: missing ` ` between `partitions` and `should`.
Since this test is checking that the stage gets finalized, we should change the name to "SPARK-40082: re-computation of shuffle map stage with no pending partitions should finalize the stage"